lib/data/argot/traject_config.rb

```diff
@@ -81,15 +82,20 @@
 
 unless settings["override"].include?("primary_isbn")
   to_field "primary_isbn" do |rec, acc|
-    Traject::MarcExtractor.cached(settings["specs"][:primary_isbn], :alternate_script => false).each_matching_line(rec) do |field, spec, extractor|
-      str = extractor.collect_subfields(field, spec).first
-      if str
-        explode = str.split
-        if(StdNum::ISBN.checkdigit(explode[0]) && !explode[1..-1].join(" ").include?("exclude"))
-          primary_isbn = explode[0]
+    Traject::MarcExtractor.cached('020q').each_matching_line(rec) do |field, spec, extractor|
```

Comment:
I don't think the logical structure here makes sense. It appears to work fine if you only have one 020 field, but if you have multiple 020 fields you get very unexpected results. This can be shown by modifying the tests.

```
# coding: utf-8
require 'spec_helper'

describe MarcToArgot do
  include Util::TrajectRunTest

  it "primary_isbn stores isbn when $q doesn't include the string 'exclude'" do
    rec = make_rec
    rec << MARC::DataField.new('020', ' ', ' ', ['a', '9789575433742'])
    rec << MARC::DataField.new('020', ' ', ' ', ['a', '7777777777777'])
    result = run_traject_on_record('ncsu', rec)
    expect(result['primary_isbn']).to eq(['9789575433742', '7777777777777'])
  end

  it "primary_isbn should be nil when $q includes the string 'exclude'" do
    rec = make_rec
    rec << MARC::DataField.new('020', ' ', ' ', ['a', '9789575433742'], ['q', 'exclude'])
    rec << MARC::DataField.new('020', ' ', ' ', ['a', '7777777777777'])
    result = run_traject_on_record('ncsu', rec)
    expect(result['primary_isbn']).to eq(['7777777777777'])
  end

  it "primary_isbn should be nil when $q includes the string 'exclude'" do
    rec = make_rec
    rec << MARC::DataField.new('020', ' ', ' ', ['a', '9789575433742 (pbk.)'], ['q', 'exclude'])
    rec << MARC::DataField.new('020', ' ', ' ', ['a', '7777777777777'])
    result = run_traject_on_record('ncsu', rec)
    expect(result['primary_isbn']).to eq(['7777777777777'])
  end

  it "primary_isbn should be nil when $q includes the string 'exclude'" do
    rec = make_rec
    rec << MARC::DataField.new('020', ' ', ' ', ['a', '7777777777777'])
    rec << MARC::DataField.new('020', ' ', ' ', ['a', '9789575433742 (pbk.)'], ['q', 'exclude'])
    result = run_traject_on_record('ncsu', rec)
    expect(result['primary_isbn']).to eq(['7777777777777'])
  end

end

```

```
Failures:

  1) MarcToArgot primary_isbn stores isbn when $q doesn't include the string 'exclude'
     Failure/Error: expect(result['primary_isbn']).to eq(['9789575433742', '7777777777777'])
     
       expected: ["9789575433742", "7777777777777"]
            got: ["9789575433742", "7777777777777", "9789575433742", "7777777777777"]
     
       (compared using ==)
     # ./spec/mta_primary_isbn_spec.rb:12:in `block (2 levels) in <top (required)>'

  2) MarcToArgot primary_isbn should be 7777777777777 when 9789575433742$q includes the string 'exclude'
     Failure/Error: expect(result['primary_isbn']).to eq(['7777777777777'])
     
       expected: ["7777777777777"]
            got: ["9789575433742", "7777777777777"]
     
       (compared using ==)
     # ./spec/mta_primary_isbn_spec.rb:20:in `block (2 levels) in <top (required)>'

  3) MarcToArgot primary_isbn should be 7777777777777 when 9789575433742$q includes the string 'exclude'
     Failure/Error: expect(result['primary_isbn']).to eq(['7777777777777'])
     
       expected: ["7777777777777"]
            got: ["9789575433742", "7777777777777"]
     
       (compared using ==)
     # ./spec/mta_primary_isbn_spec.rb:28:in `block (2 levels) in <top (required)>'

  4) MarcToArgot primary_isbn should be 7777777777777 when 9789575433742$q includes the string 'exclude'
     Failure/Error: expect(result['primary_isbn']).to eq(['7777777777777'])
     
       expected: ["7777777777777"]
            got: ["7777777777777", "9789575433742"]
     
       (compared using ==)
     # ./spec/mta_primary_isbn_spec.rb:36:in `block (2 levels) in <top (required)>'

Finished in 9.61 seconds (files took 0.68616 seconds to load)
940 examples, 4 failures, 32 pending

```